src/rulesets/Base/base.jl

```diff
@@ -91,6 +92,12 @@
 @scalar_rule(fma(x, y, z), (y, x, One()))
 @scalar_rule(max(x, y), @setup(gt = x > y), (gt, !gt))
 @scalar_rule(min(x, y), @setup(gt = x > y), (!gt, gt))
+@scalar_rule(clamp(x, low, high),
+    @setup(
+        islow = x < low,
+        ishigh = high < x),
+        (!( islow | ishigh), islow, ishigh)
+)
```

Comment:
Sorry, I was missing a comma before:
```suggestion
@scalar_rule(
    clamp(x, low, high),
    @setup(
        islow = x < low,
        ishigh = high < x,
    ),
    (!(islow | ishigh), islow, ishigh),
)
```

Reply:
Great i was puzzled why it failed.